src/query/service/src/interpreters/interpreter.rs

```diff
@@ -71,6 +75,7 @@ pub trait Interpreter: Sync + Send {
 
         let query_ctx = ctx.clone();
         build_res.main_pipeline.set_on_finished(move |may_error| {
+            InterpreterMetrics::record_query_finished(&query_ctx, may_error.clone());
```

Comment:
I think this is better. Because there maybe have an error when the pipeline set_on_finished.
```suggestion
match may_error {
    None => {
        InterpreterMetrics::record_query_finished(&query_ctx, may_error.clone());
        Ok(())
    },
    Some(error) => {
        InterpreterMetrics::record_query_error(&ctx);
        Err(error.clone())
    },
}
```

Reply:
so the stats for failed query would not be accumulated?

Reply:
Ohh, ignore this. I see it already record a query failed in record_query_finished